packages/patternfly-4/react-core/src/components/Select/SingleSelect.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Select/select.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** Content rendered inside the SingleSelect */
+  children: PropTypes.node.isRequired,
+  /** Additional classes added to the SingleSelect control */
+  className: PropTypes.string,
+  /** Flag indicating the Select is expanded */
+  isExpanded: PropTypes.bool,
+  /** Currently selected option */
+  selected: PropTypes.string,
+  /** Additional props are spread to the container <select> */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  className: '',
+  isExpanded: false,
+  selected: ''
+};
+
+class SingleSelect extends React.Component {
+  refCollection = [];
+
+  componentDidMount() {
+    if (this.props.openedOnEnter) {
+      this.refCollection[0].current.focus();
+    }
+  }
+
+  extendChildren() {
+    return React.Children.map(this.props.children, (child, index) =>
+      React.cloneElement(child, {
+        selected: this.props.selected === child.props.label,
+        sendRef: this.sendRef,
+        keyHandler: this.keyHandler,
+        index
+      })
+    );
+  }
+
+  sendRef = (ref, index) => {
+    this.refCollection[index] = ref;
+  };
+
+  keyHandler = (index, position) => {
```

Comment:
We may want to move this into a utility class if the other variations of select may use it. Also maybe ad some unit test.

Reply:
I'll refactor this as we add more variants if its needed.

Reply:
Can this use the keyHandler function that Rebecca added to the helpers/util.ts file for AppLauncher and Dropdown to use?